File: app/src/main/java/com/philkes/pin2pdf/storage/local/dao/PinDao.java
package com.philkes.pin2pdf.storage.local.dao;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import com.philkes.pin2pdf.storage.local.entity.Pin;

import java.util.List;

@Dao
public interface PinDao {

    @Query("SELECT * FROM Pin")
    List<Pin> getAll();

    @Query("SELECT * FROM pin WHERE pinId IN (:pinIds) ORDER BY pinId DESC")
    List<Pin> loadAllByPinIds(List<String> pinIds);

    @Insert
    void insertAll(List<Pin> pins);

    @Delete
    void delete(Pin pin);

    @Query("DELETE FROM Pin")
    void clear();

    @Update
    void update(Pin pin);


}

File: app/src/main/java/com/philkes/pin2pdf/storage/local/dao/PinDao.kt
package com.philkes.pin2pdf.storage.local.dao

import androidx.room.*
import com.philkes.pin2pdf.storage.local.entity.Pin

@Dao
interface PinDao {
    @get:Query("SELECT * FROM Pin")
    val all: List<Pin?>?

    @Query("SELECT * FROM pin WHERE pinId IN (:pinIds) ORDER BY pinId DESC")
    fun loadAllByPinIds(pinIds: List<String>): List<Pin>

    @Insert
    fun insertAll(pins: List<Pin>?)

    @Delete
    fun delete(pin: Pin?)

    @Query("DELETE FROM Pin")
    fun clear()

    @Update
    fun update(pin: Pin?)
}